### ISSUE TITLE
Build fails when trying to add IJulia package

### ISSUE THREAD

**Issue by edouardruiz** [JuliaLang/IJulia.jl#389]
Hi,

I'm trying to install IJulia kernel for Jupyter on Mac Os (10.11.1 but it keeps failing when building Homebrew.

When I run ``Pkg.add("IJulia")`` in a Julia console, I get the following output:
```
INFO: Initializing package repository /Users/edouard/.julia/v0.4
INFO: Cloning METADATA from git://github.com/JuliaLang/METADATA.jl
INFO: Installing BinDeps v0.3.19
INFO: Installing Compat v0.7.7
INFO: Installing Conda v0.1.8
INFO: Installing Homebrew v0.2.0
INFO: Installing IJulia v1.1.8
INFO: Installing JSON v0.5.0
INFO: Installing Nettle v0.2.0
INFO: Installing SHA v0.1.2
INFO: Installing URIParser v0.1.1
INFO: Installing ZMQ v0.3.0
INFO: Building Homebrew
Warning: New definition 
    convert(Type{Compat.LazyLinSpace.LinSpace{T}},Compat.LazyLinSpace.LinSpace{T<:FloatingPoint}) at /Users/edouard/.julia/v0.4/Compat/src/linspace.jl:189
is ambiguous with: 
    convert(Type{Compat.LazyLinSpace.LinSpace{T}},Compat.LazyLinSpace.LinSpace{T}) at /Users/edouard/.julia/v0.4/Compat/src/linspace.jl:188.
To fix, define 
    convert(Type{Compat.LazyLinSpace.LinSpace{T<:FloatingPoint}},Compat.LazyLinSpace.LinSpace{_<:FloatingPoint})
before the new definition.
==========================================[ ERROR: Homebrew ]===========================================

LoadError: LoadError: LoadError: LoadError: LoadError: LoadError: UndefVarError: __precompile__ not defined
while loading /Users/edouard/.julia/v0.4/URIParser/src/URIParser.jl, in expression starting on line 1
while loading /Users/edouard/.julia/v0.4/BinDeps/src/dependencies.jl, in expression starting on line 217
while loading /Users/edouard/.julia/v0.4/BinDeps/src/BinDeps.jl, in expression starting on line 507
while loading /Users/edouard/.julia/v0.4/Homebrew/src/bindeps_integration.jl, in expression starting on line 2
while loading /Users/edouard/.julia/v0.4/Homebrew/src/Homebrew.jl, in expression starting on line 381
while loading /Users/edouard/.julia/v0.4/Homebrew/deps/build.jl, in expression starting on line 1

========================================================================================================
INFO: Building Nettle
===========================================[ ERROR: Nettle ]============================================

LoadError: UndefVarError: provides not defined
while loading /Users/edouard/.julia/v0.4/Nettle/deps/build.jl, in expression starting on line 15

========================================================================================================
INFO: Building ZMQ
=============================================[ ERROR: ZMQ ]=============================================

LoadError: UndefVarError: provides not defined
while loading /Users/edouard/.julia/v0.4/ZMQ/deps/build.jl, in expression starting on line 8

========================================================================================================
INFO: Building IJulia
===========================================[ ERROR: IJulia ]============================================

LoadError: LoadError: LoadError: UndefVarError: provider not defined
while loading /Users/edouard/.julia/v0.4/Conda/src/bindeps_conda.jl, in expression starting on line 40
while loading /Users/edouard/.julia/v0.4/Conda/src/Conda.jl, in expression starting on line 238
while loading /Users/edouard/.julia/v0.4/IJulia/deps/build.jl, in expression starting on line 2

========================================================================================================

============================================[ BUILD ERRORS ]============================================

WARNING: IJulia, Homebrew, Nettle and ZMQ had build errors.

 - packages with build errors remain installed in /Users/edouard/.julia/v0.4
 - build the package(s) and all dependencies with `Pkg.build("IJulia", "Homebrew", "Nettle", "ZMQ")`
 - build a single package by running its `deps/build.jl` script

========================================================================================================
INFO: Package database updated
```

I'm a newbie with Julia and was more trying to install it for test purposes, so I'm not sure what to do. I already tried removing the package directory using ``rm(Pkg.dir(),recursive=true)``but it didn't help me.

Any idea of what could be wrong? Is it a pure Homebrew issue?

**Comment by stevengj** [comment]
The `__precompile__ not defined` error is bizarre.  `__precompile__` is a built-in function in Julia 0.4.

What version of Julia are you using?  What is the output of `versioninfo()`?

**Comment by edouardruiz** [comment]
Here is the output of ``versioninfo()``:

```
Julia Version 0.4.0-dev+3631
Commit 949f202* (2015-02-28 19:14 UTC)
Platform Info:
  System: Darwin (x86_64-apple-darwin14.1.0)
  CPU: Intel(R) Core(TM) i5-4308U CPU @ 2.80GHz
  WORD_SIZE: 64
  BLAS: libopenblas (DYNAMIC_ARCH NO_AFFINITY Haswell)
  LAPACK: libopenblas
  LIBM: libopenlibm
  LLVM: libLLVM-3.3
```

And indeed I can't call ``__precompile__`` in a console:
```
ERROR: UndefVarError: __precompile__ not defined
```

**Comment by JobJob** [comment]
Have you tried installing the latest julia release?

**Comment by stevengj** [comment]
Yes, you need a released version of Julia; your 0.4 pre-release version is ancient.

**Comment by stevengj** [comment]
(This is a bug in the `REQUIRE` file for URIParser.jl.)
